framework/Models/HybridModels/LogicalModel.py

```diff
@@ -0,0 +1,231 @@
+        elif set(romInpVars) != set(inpVars) or set(romOutVars) != set(outVars):
+          self.raiseAnError(IOError,'ROM "{}" does not have the same Features and Targets as ROM "{}"!'.format(modelName, romName))
+      elif modelInst.type == 'Code':
+        self.raiseAWarning('The input/output consistency check is not performed for Model "{}" among {}!'.format(modelName, self.name))
+      else:
+        self.raiseAnError(IOError, 'Model "{}" with type "{}" can not be accepted by {}!'.format(modelName, modelInst.type, self.name))
+    if extModelVars is not None and romInpVars is not None:
+      romVars = romInpVars + romOutVars
+      if set(romVars) != set(extModelVars):
+        self.raiseAnError(IOError,'"Variables" provided to model "{}" are not the same as ROM "{}"!'.format(extModelName, romName),
+                          'The variables listed in "Target" and "Features" of ROM should be also listed for ExternalModel under "Variables" node!')
+    if extModelVars is not None or romInpVars is not None:
+      if not inputVars:
+        modelName = romName if romInpVars else extModelName
+        self.raiseAnError(IOError, 'An input DataObject is required for {}!'.format(self.name),
+                          'This is because model "{}" expects a DataObject as input.'.format(modelName))
+      else:
+        error = set(inputVars) - set(romInpVars) if romInpVars else set(inputVars) - set(extModelVars)
+        modelName = romName if romInpVars else extModelName
+        if len(error) > 0:
+          self.raiseAnError(IOError, 'Variable(s) listed under DataObject "{}" could not be find in model "{}"!'.format(','.join(error), modelName))
+
+  def createNewInput(self, myInput, samplerType, **kwargs):
+    """
+      This function will return a new input to be submitted to the model, it is called by the sampler.
+      @ In, myInput, list, the inputs (list) to start from to generate the new one
+      @ In, samplerType, string, is the type of sampler that is calling to generate a new input
+      @ In, **kwargs, dict,  is a dictionary that contains the information coming from the sampler,
+           a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
+      @ Out, newInputs, dict, dict that returns the new inputs for each sub-model
+    """
+    self.raiseADebug("{}: Create new input.".format(self.name))
+    # TODO: standardize the way to handle code/external model/rom inputs
+    modelToRun = self.controlFunction.evaluate("evaluate", kwargs)
+    if modelToRun not in self.modelInstances:
+      self.raiseAnError(IOError, 'Model (i.e. {}) returned from "ControlFunction" is not valid!'.format(modelToRun),
+                        'Available models are: {}'.format(','.join(self.modelInstances.keys())))
+    kwargs['modelToRun'] = modelToRun
+    if self.modelInstances[modelToRun].type == 'Code':
+      codeInput = []
+      for elem in myInput:
+        if isinstance(elem, Files.File):
```

Comment:
copied from previous comment: 
```
Are you fine if I work on this issue in another pull request? Since the changes will also affect EnsembleModel and CodeInterface Entities. It may take longer and further discussion on the solution for this issue.
```